drivers/hmis/app/models/hmis/form/definition.rb

```diff
@@ -268,6 +268,20 @@ class Hmis::Form::Definition < ::GrdaWarehouseBase
     where(identifier: instance_scope.pluck(:definition_identifier))
   end
 
+  CURRENT_VERSION_SQL = <<~SQL.freeze
+    SELECT DISTINCT ON (identifier) id
+    FROM hmis_form_definitions
+    WHERE status IN ('published', 'draft')
+    ORDER BY identifier, CASE
+      WHEN status = 'published' THEN 1
+      WHEN status = 'draft' THEN 2
+      ELSE 3
+    END, version DESC
+  SQL
+  scope :current_versions, -> do
```

Comment:
This scope probably warrants an explanatory comment. I could see the name `current_versions` being misinterpreted as _active/published versions_ that can be used in the UI, which it's not (since some are draft).

Reply:
Agree this needs a comment, it's even feeling to me like we should consider renaming `current_versions` entirely. A couple ideas:
- `one_per_identifier`
- `per_identifier`
- `identifier_representatives`
- `representative_versions` * my favorite

Reply:
Hm. Another idea is to scrap the complicated logic relating to statuses, and just simply have this scope be called `latest_versions` and return the latest version (by version number) for each form.

That makes the code and variable naming simpler. It also enables us to use the `one_for_column` arel helper Elliot mentioned above, without modification. 

The result would be that in the frontend table of form definitions, if a user changes a form's title/role/etc., then that change would still show up in the table before the user has hit publish.

Reply:
ok huddle summary>>
* use latest_version as suggested
* gql type handles choosing which title to display
* don't worry about search scope for now
* use data loaders for versions
* use identifier as cache key